src/Hal/Bundle/PhpMetricsCollector/Collector/PhpMetricsCollector.php

```diff
@@ -1,101 +1,122 @@
 <?php
+
 namespace Hal\Bundle\PhpMetricsCollector\Collector;
 
-use Hal\Component\Token\Tokenizer;
-use Hal\Component\Token\TokenType;
-use Hal\Metrics\Complexity\Component\McCabe\McCabe;
-use Hal\Metrics\Complexity\Text\Halstead\Halstead;
-use Hal\Metrics\Complexity\Text\Length\Loc;
-use Hal\Metrics\Design\Component\MaintainabilityIndex\MaintainabilityIndex;
+use Hal\Application\Analyze;
+use Hal\Application\Config\Config;
+use Hal\Component\Issue\Issuer;
+use Hal\Component\Output\TestOutput;
+use Hal\Metric\ClassMetric;
+use Hal\Metric\Consolidated;
+use Hal\Metric\InterfaceMetric;
+use Hal\Metric\Metric;
+use Hal\Violation\Violations;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+use Symfony\Component\HttpKernel\Kernel;
 
 class PhpMetricsCollector extends DataCollector
 {
-    public function collect(Request $request, Response $response, \Exception $exception = null)
+    /** @var Kernel */
+    private $kernel;
```

Comment:
I think that now Symfony provide a way to inject the project dir in a env variable, so you wouldn't need to inject the kernel here.